pyam/core.py

```diff
@@ -601,11 +601,10 @@ def categorize(self, name, value, criteria,
                           ('linestyle', linestyle)]:
             if arg:
                 run_control().update({kind: {name: {value: arg}}})
-
         # find all data that matches categorization
-        rows = _apply_criteria(self.data, criteria,
+        rows = _apply_criteria(self._data, criteria,
                                in_range=True, return_test='all')
-        idx = _meta_idx(rows)
+        idx = _meta_idx(rows.index.to_frame())
```

Comment:
actually, the function `_meta_idx` seems to do the same as `_make_idx` (but less efficient and less flexible)

Reply:
okay, will switch to `_make_index`
i guess `_meta_idx` can be removed from core.py eventually?

Reply:
yes, it should be replaced by `_make_idx` eventually in all places, but not as part of this PR - only in functions that are touched in the PR